Forge.TreeWalker/src/TreeWalkerParameters.cs

```diff
@@ -126,55 +128,57 @@ public class TreeWalkerParameters
 
         /// <summary>
         /// Instantiates a TreeWalkerParameters object with the properties that are required to instantiate a TreeWalkerSession object.
+        /// Note: Recommended to cache the deserialized ForgeTree in your app and use that to avoid Forge deserializing the JsonSchema each initialize.
         /// </summary>
         /// <param name="sessionId">The unique identifier for this session.</param>
-        /// <param name="jsonSchema">The JSON schema.</param>
+        /// <param name="forgeTree">The ForgeTree for this session.</param>
         /// <param name="forgeState">The Forge state.</param>
         /// <param name="callbacks">The callbacks object.</param>
         /// <param name="token">The cancellation token.</param>
         public TreeWalkerParameters(
             Guid sessionId,
-            string jsonSchema,
+            ForgeTree forgeTree,
             IForgeDictionary forgeState,
             ITreeWalkerCallbacks callbacks,
             CancellationToken token)
         {
             if (sessionId == Guid.Empty) throw new ArgumentNullException("sessionId");
-            if (string.IsNullOrWhiteSpace(jsonSchema)) throw new ArgumentNullException("jsonSchema");
+            if (forgeTree == null) throw new ArgumentNullException("forgeTree");
             if (forgeState == null) throw new ArgumentNullException("forgeState");
             if (callbacks == null) throw new ArgumentNullException("callbacks");
             if (token == null) throw new ArgumentNullException("token");
 
             this.SessionId = sessionId;
-            this.JsonSchema = jsonSchema;
+            this.ForgeTree = forgeTree;
```

Comment:
just to confirm, looks like you just switched line 140-152 with line 169-181. Besides these, only some comments are changed, right?

Reply:
I swapped the constructors to put the ForgeTree one on top. Just because it is preferred to use that one. No functional changes.

---
In reply to: [470868342](https://github.com/microsoft/Forge/pull/25#discussion_r470868342) [](ancestors = 470868342)